examples/oidc_cli/src/main.rs

```diff
@@ -280,9 +281,33 @@ impl OidcCli {
         let StoredSession { client_session, user_session, client_credentials } =
             serde_json::from_str(&serialized_session)?;
 
+        println!("homeserver = {}", client_session.homeserver);
+        let (homeserver, insecure) =
+            if let Some(base) = client_session.homeserver.strip_prefix("http://") {
+                (base, true)
+            } else {
+                (
+                    client_session
+                        .homeserver
+                        .strip_prefix("https://")
+                        .unwrap_or(&client_session.homeserver),
+                    false,
+                )
+            };
+        let homeserver = homeserver.strip_suffix('/').unwrap_or(homeserver);
+        let server_name = ServerName::parse(homeserver)?;
+
         // Build the client with the previous settings from the session.
-        let client = Client::builder()
-            .homeserver_url(client_session.homeserver)
+        let mut client = Client::builder();
+
+        if insecure {
+            client = client.insecure_server_name_no_tls(&server_name);
+        } else {
+            client = client.server_name(&server_name);
+        }
+
+        let client = client
+            .handle_refresh_tokens()
```

Comment:
If this is activated, we need to watch for tokens changes to persist the new tokens.